flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferAvailabilityListener.java

```diff
@@ -34,11 +30,8 @@
 	void notifyDataAvailable();
 
 	/**
-	 * Allows the listener to react to a priority event before it is added to the outgoing buffer queue.
-	 *
-	 * @return true if the event has been fully processed and should not be added to the buffer queue.
+	 * Called when the first priority event is added to the head of the buffer queue.
 	 */
-	default boolean notifyPriorityEvent(BufferConsumer eventBufferConsumer) throws IOException {
-		return false;
+	default void notifyPriorityEvent() {
```

Comment:
could you elaborate a bit more in the commit message, what has been simplified, why and what are the benefits?

Reply:
Added a few thoughts. Let me know if it makes things clearer.